biomage/utils/data.py

```diff
@@ -5,7 +5,79 @@
 from ..utils.constants import PRODUCTION, STAGING
 
 
-def modified_records(item, target_table, config):
+def add_env_user_to_experiment(username, config):
+    """
+    Add cognito userId of the current user to the experiment.
+    This function should return a string that is the current user id.
+    """
+
+    client = boto3.client('cognito-idp')
+
+    try:
+        user = client.admin_get_user(
+            UserPoolId=config["user-pool-id"],
+            Username=username
+        )
+
+        return user["Username"]
+
+    except Exception as e:
+        click.echo(
+            f"failed to get userId to add into RBAC \
+            with exception: \n {e}"
+        )
+
+
+def remap_sample_references(samples, sandbox_id):
+    """
+    Prefix sandbox_id to references in samples
+    """
+
+    return {
+        "M" : {
+            f"{sandbox_id}-{sample_id}": {
+                "M": {
+                    **samples['M'][sample_id]['M'],
+                    "files": remap_file_references(
+                        samples['M'][sample_id]["M"]["files"],
+                        sandbox_id
+                    ),
+                    "uuid": {"S" : f"{sandbox_id}-{sample_id}"},
+                    "projectUuid": { 
+                        "S" : f"{sandbox_id}-{samples['M'][sample_id]['M']['projectUuid']['S']}"
+                    }
+                },
+            } for sample_id in samples['M']
+        }
+    }
+
+
+def remap_file_references(files, sandbox_id):
+    """
+    Prefix sandbox_id to references in files
+    """
+
+    valid_filenames = [ file for file in files["M"] if file != "lastModified" ]
+
+    remapped_files = {
+        "M" : {
+            file : {
+                "M": {
+                    **files['M'][file]['M'],
+                    "path": {
+                        "S": f"{sandbox_id}-{files['M'][file]['M']['path']['S']}"
+                    }
+                }
+            } for file in valid_filenames
```

Comment:
again having this list comprehension nested in a dictionary creation just decrease legibility unnecessarily. Moreover, you are not modifying any key in this dict so this whole thing could be simplified to the following snippet. In the previous function `remap_sample_references` rebuild the whole dict is kind of acceptable as you need to modify the keys but here it's not.
```
for file in filenames:
    files['M'][file]['M']['path']['S'] = f"{sandbox_id}-{files['M'][file]['M']['path']['S']}"
```

Reply:
Chagned to for loop